worlds/spelunker/Rules.py

```diff
@@ -0,0 +1,121 @@
+
+    set_rule(world.multiworld.get_location("B30F - Mound Item", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 4) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("B30F - Far Bottom Left", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+    set_rule(world.multiworld.get_location("B29F - Left Wall Rope Ledge", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+    set_rule(world.multiworld.get_location("B29F - Left Pit Item", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("B29F - Bombable Wall", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 6) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("B29F - Left Wall Item", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("B28F - Locked Item 1", player), lambda state: state.has("Red Key", player, 5) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+    set_rule(world.multiworld.get_location("B28F - Locked Item 2", player), lambda state: state.has("Red Key", player, 5) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("B28F - Final Rope Left", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+    set_rule(world.multiworld.get_location("B28F - Final Rope Right", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 5) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("B28F - Final Boulder", player), lambda state: state.has("Red Key", player, 4) and state.has("Dynamite", player, 6) and state.has("Blue Key", player, 5))
+
+    set_rule(world.multiworld.get_location("Golden Pyramid", player), lambda state: state.has("Red Key", player, 5) and state.has("Dynamite", player, 6) and state.has("Blue Key", player, 6))
+
+    hidden_rules(world)
+    
+
+def hidden_rules(world: "SpelunkerWorld") -> None:
+    player = world.player
+    if not world.options.hidden_items:
+        return
+    set_rule(world.multiworld.get_location("B1F - Side Shaft Hidden Item", player), lambda state: state.has("Dynamite", player, 1))
+    set_rule(world.multiworld.get_location("B8F - Boulder Pit Hidden Item", player), lambda state: state.has("Dynamite", player, 2))
+    set_rule(world.multiworld.get_location("B16F - Blue Door Hidden Item", player), lambda state: state.has("Blue Key", player, 4))
```

Comment:
```suggestion
    set_rule(world.multiworld.get_location("Golden Pyramid", player), lambda state: state.has("Red Key", player, 5) and state.has("Dynamite", player, 6) and state.has("Blue Key", player, 6))

    if world.options.hidden_items:
        set_rule(world.multiworld.get_location("B1F - Side Shaft Hidden Item", player), lambda state: state.has("Dynamite", player, 1))
        set_rule(world.multiworld.get_location("B8F - Boulder Pit Hidden Item", player), lambda state: state.has("Dynamite", player, 2))
        set_rule(world.multiworld.get_location("B16F - Blue Door Hidden Item", player), lambda state: state.has("Blue Key", player, 4))
```

Unless there's some specific reason you wanted to keep hidden_rules as a separate function, would recommend just rolling it into set_rules

Reply:
I was thinking about it similarly to Yoshi’s Island extra rules, except I didn’t realize going in that it was only 8 locations. Will change that